linux/ibus-keyman/src/engine.c

```diff
@@ -769,13 +832,24 @@ ibus_keyman_engine_process_key_event(
   guint state
 ) {
   IBusKeymanEngine *keyman = (IBusKeymanEngine *)engine;
+  keyman->commit_item->keyval     = keyval;
+  keyman->commit_item->keycode    = keycode;
+  keyman->commit_item->state      = state;
 
   gboolean isKeyDown = !(state & IBUS_RELEASE_MASK);
 
   g_message("-----------------------------------------------------------------------------------------------------------------");
   g_message(
-      "DAR: ibus_keyman_engine_process_key_event - keyval=0x%02x keycode=0x%02x, state=0x%02x, isKeyDown=%d", keyval, keycode,
-      state, isKeyDown);
+      "DAR: ibus_keyman_engine_process_key_event - keyval=0x%02x keycode=0x%02x, state=0x%02x, isKeyDown=%d, supports_prefilter=%d", keyval, keycode,
+      state, isKeyDown, engine->client_capabilities & IBUS_CAP_PREFILTER);
+
+  // This keycode is a fake keycode that we send when it's time to commit the text, ensuring the
+  // correct output order of backspace and text.
+  if ((engine->client_capabilities & IBUS_CAP_PREFILTER) && keycode == KEYMAN_F24_KEYCODE_OUTPUT_SENTINEL &&
```

Comment:
Good thing we check `keycode` and not `keyval` since Chromium sends us a different value than what we used in the forward key event (from Chromium: 0x1008ff43, we sent: 0x100fdd0). Other apps send us the same value back.